File: webv2/summary/summary.go
package summary

import (
	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal/reports"
)

type ConversionSummary struct {
	SrcTable         string
	SpTable          string
	Errors           []reports.Issue
	Warnings         []reports.Issue
	Suggestions      []reports.Issue
	Notes            []reports.Issue
	ErrorsCount      int
	WarningsCount    int
	SuggestionsCount int
	NotesCount       int
}

File: ui/src/app/model/summary.ts
import { IIssue } from './structured-report'

export default interface ISummary {
  SrcTable: string
  SpTable: string
  Errors: IIssue[]
  Warnings: IIssue[]
  Suggestions: IIssue[]
  Notes: IIssue[]
  ErrorsCount: number
  WarningsCount: number
  SuggestionsCount: number
  NotesCount: number
}
export interface ISummaryRow {
  type: string
  content: string
  isRead: boolean
}
